src/main/java/io/github/debuggyteam/architecture_extensions/blocks/PostLanternBlock.java

```diff
@@ -9,31 +10,56 @@
 import net.minecraft.state.StateManager;
 import net.minecraft.state.property.BooleanProperty;
 import net.minecraft.state.property.Properties;
+import net.minecraft.util.function.BooleanBiFunction;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
 import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.util.shape.VoxelShapes;
 import net.minecraft.world.BlockView;
 import net.minecraft.world.WorldAccess;
 
 public class PostLanternBlock extends Block {
 	public static final BooleanProperty WATERLOGGED = Properties.WATERLOGGED;
-
-	protected static final VoxelShape POST_CAP_SHAPE = Block.createCuboidShape(5.0, 0.0, 5.0, 11.0, 6.0, 11.0);
+	public static final BooleanProperty HANGING = Properties.HANGING;
+	
+	private static final VoxelShape LANTERN_BODY = Block.createCuboidShape(5, 0, 5, 11, 6, 11);
+	private static final VoxelShape LANTERN_CAP = Block.createCuboidShape(4, 5, 4, 12, 6, 12);
+	private static final VoxelShape LANTERN_FINIAL = Block.createCuboidShape(6, 6, 6, 10, 7, 10);
```

Comment:
Found a typo 🍍

Reply:
finial is the correct spelling.

Reply:
oh, i thought it was a typo of "final" lol